torch_nets/models.py

```diff
@@ -31,59 +45,116 @@ class ANN(Module):  # pylint: disable=too-many-instance-attributes
-        self.lin_drop2 = Dropout(dropout)
-        self.lin_drop3 = Dropout(dropout)
-        self.lin_drop4 = Dropout(dropout)
-
-    def forward(self, batch):
-        """Pass ``batch`` through the model.
+        """Initialize the ANN model."""
+        dims = [n_in] + [neurons] * (n_layers - 1) + [n_out]
+        layers = []
+
+        for i in range(n_layers):
+            layers.append(nn.Linear(dims[i], dims[i + 1]))
+            if i < n_layers - 1:
+                layers.append(nn.ReLU())  # type: ignore
+                layers.append(nn.Dropout(dropout))  # type: ignore
+
+        super().__init__(*layers)
+
+        fmean = fstd = omean = ostd = None
+
+        if features_mean is not None:
+            assert features_std is not None
+            assert len(features_mean) == len(features_std)
+            fmean = torch.tensor(features_mean)
+            fstd = torch.tensor(features_std)
+
+        if outputs_mean is not None:
+            assert outputs_std is not None
+            assert len(outputs_mean) == len(outputs_std)
+            if output_groups is None:
+                omean = torch.tensor(outputs_mean)
+                ostd = torch.tensor(outputs_std)
+            else:
+                assert len(output_groups) == len(outputs_mean)
+                omean = torch.tensor(
+                    [x for x, g in zip(outputs_mean, output_groups) for _ in range(g)]
+                )
+                ostd = torch.tensor(
+                    [x for x, g in zip(outputs_std, output_groups) for _ in range(g)]
+                )
+
+        self.register_buffer("features_mean", fmean)
+        self.register_buffer("features_std", fstd)
+        self.register_buffer("outputs_mean", omean)
+        self.register_buffer("outputs_std", ostd)
+
+        self.to(torch.device(device))
+
+    def forward(self, input: Tensor):  # pylint: disable=redefined-builtin
+        """Pass the input through the model.
 
         Parameters
         ----------
-        batch : Tensor
+        input : Tensor
             A mini-batch of inputs.
 
         Returns
         -------
         Tensor
-            The result of passing ``batch`` through the model.
+            The model output.
 
         """
-        batch = F.relu(self.linear1(batch))
-        batch = self.lin_drop1(batch)
+        if self.features_mean is not None:
+            input = (input - self.features_mean) / self.features_std
 
-        batch = F.relu(self.linear2(batch))
-        batch = self.lin_drop2(batch)
+        # pass the input through the layers
+        output = super().forward(input)
 
-        batch = F.relu(self.linear3(batch))
-        batch = self.lin_drop3(batch)
+        if self.outputs_mean is not None:
+            output = output * self.outputs_std + self.outputs_mean
 
-        batch = F.relu(self.linear4(batch))
-        batch = self.lin_drop4(batch)
+        return output
 
-        batch = self.linear5(batch)
+    def load(self, path: str) -> "ANN":
```

Comment:
```suggestion
    def load(self, path: str) -> ANN:
```

I don't think this needs to be in quotes as we declare the class above.